### ISSUE TITLE
[Neat URL] trkid= not cleaned from Netflix links when codex.nflxext.com is blocked

### ISSUE THREAD

**Issue by vertigo220** [Smile4ever/firefoxaddons#134]
I've just copy/pasted the same issue I reported for ClearURLs. Since both addon's are doing the same thing, maybe it's just how they work, but that would be disappointing. Anyways, here it is:

Description

Not sure if this is a bug or not, but codex.nflxext.com script has to be allowed in order for a trkid Netflix link to be cleaned. For example, with uBo and uMatrix installed, going to https://www.netflix.com/title/80166314?trkid=1234567&MSG_TITLE=80166314&lnktrk=EMP&g=6E5416B8D93DD75DB84131269FB4725A6455A8C2&lkid=W2W_ROW_1_MDP_ will only convert the link to https://www.netflix.com/title/80166314 once it has been allowed. It seems the conversion shouldn't be reliant on requiring script to run on the destination page.

I was going to make a separate issue for the fact that google search results aren't cleaned by ClearURLs, but I now suspect it's likely due to the same cause. It seems very counter-productive to globally require 3rd-party scripts just to clean links, so I'm hoping this is just a bug and not a requirement for it to function.

Steps to Reproduce

1) Install uBo and uMatrix and set to block 3rd-party (alternately, you could just use uBo and forego uMatrix, as the result would be the same, since uMatrix basically just allows more fine-grained control)
2) Go to above link and notice it's not cleaned
3) Allow the above in uBo and, if applicable, allow scripts for it in uMatrix, then go to the link again and notice that it's cleaned

Platform: Win10x64

**Comment by vertigo220** [comment]
It opens for me in Waterfox, Firefox, Brave, and Vivaldi, so not sure why it's not working for you.  Are you sure it's using the whole thing when you click on it, since it's split across multiple lines? Interestingly, Brave and Vivaldi "clean" the link on their own, without addons. Of course, I have no idea if, when this is happening and when codex.nflxext.com is unblocked, the link is being cleaned _before_ going to it or after.

**Comment by crssi** [comment]
Tried with EDGE and IE.

On both I get; Netflix Site Error - Page Not Found
Maybe is GEO sensitive.

**Comment by vertigo220** [comment]
That's possible. Any difference with this one?

https://www.netflix.com/title/80166314?trkid=13710079&MSG_TITLE=80166314&lnktrk=EMP&g=DB0C85746BCE5A0B31CBE2CC588993CA64522830&lkid=TITLE1_RECADD_MDP

**Comment by Smile4ever** [comment]
For me both links give 404 too, but the cleaning shouldn't be different then.

I'm having trouble to follow the steps you mentioned.

1. I installed uBlock Origin and uMatrix. How do I configure it to deny third party scripts?
* I tried to configure it in uBlock Origin with this filter: `||nflxext.com`. Seems to work:
![image](https://user-images.githubusercontent.com/544411/39771719-77cb4e20-52f3-11e8-83bc-6a333f2e234b.png)


* I also tried adding `netflix.com odex.nflxext.com script block` to uMatrix. Not sure whether it did the right thing.
![image](https://user-images.githubusercontent.com/544411/39771783-9e832218-52f3-11e8-8197-e9dd1ff608dc.png)

2. For me, it was always cleaned. I'm not sure I configured the addons well, though. Please see above.

3. The link was always cleaned for me.

**Comment by vertigo220** [comment]
I'd suggest just using uBo without uMatrix, since for the purposes of this example it will have the same effect. Open uBo's settings and check "I am an advanced user" then open the uBo panel and click the right side of the left column in the 3rd-party scripts and 3rd-party frames rows. That will mark both columns in both those rows red, which means it's blocking those (the left column is global, the right column is the current domain). Then load the link (or I suppose a Netflix link that works for you; if you have a Netflix subscription and have an email from them, click a link from there) and when you open the uBo panel there should be a minus sign in the right column in the assets.nflxext.com and codex.nflxext.com rows and the link should still have the tracking info in it.

**Comment by Smile4ever** [comment]
I followed everything, but I could not reproduce that it is not cleaning the links. It's always cleaning the links for me. Can I have a screenshot of your Neat URL configuration?

**Comment by vertigo220** [comment]
![1](https://user-images.githubusercontent.com/12024557/39784391-381c6352-52e6-11e8-9c3d-9de87b3391db.png)
![2](https://user-images.githubusercontent.com/12024557/39784400-3ebe1660-52e6-11e8-9476-2935693d4554.png)

If this works similarly to ClearURLs, the problem may be that it's not being cleaned since there's not a request being made by it due to it being blocked. See my issue for that addon [here](https://github.com/KevinRoebert/ClearUrls/issues/40#issuecomment-387505430).

**Comment by Smile4ever** [comment]
I don't see why the script should be required to CLEAN the link. I have seen cases where scripts prevent links from being cleaned, but never the other way around.

I don't see trkid added in your screenshot.

If you want me to look closer into this issue, I will need a step-by-step instruction with screenshots. Also try logging out of Netflix (see if that makes a difference, since I'm not logged in).

What do you get in the browser console if you enable the logging of Neat URL?

**Comment by Smile4ever** [comment]
This is now tracked in https://github.com/Smile4ever/Neat-URL/issues/134
